OnlineShop.Tests/Controllers/CategoriesControllerTests.cs

```diff
@@ -0,0 +1,142 @@
+        }
+
+        [TestMethod]
+        public void PutCategoryNullCategoryTest()
+        {
+            //arrange
+            var category = null as Category;
+
+            //act
+            var actual = _target.PutCategory(_fixture.Create<int>(), category);
+
+            //assert
+            (actual as StatusCodeResult)?.StatusCode.Should().Be(400, "Status code is not NoContent");
+        }
+
+        [TestMethod]
+        public void PutCategoryWrongCategoryIdTest()
+        {
+            //arrange
+            var category = _fixture.Create<Category>();
+
+            //act
+            var actual = _target.PutCategory(_fixture.Create<int>(), category);
+
+            //assert
+            (actual as StatusCodeResult)?.StatusCode.Should().Be(400, "Status code is not NoContent");
+        }
+
+        [TestMethod]
+        public void PutCategoryOptimisticExceptionTest()
+        {
+            //arrange
+            var category = _fixture.Create<Category>();
+            var exception = new DbUpdateConcurrencyException();
+
+            _uow.Setup(x => x.CategoryRepository).Returns(_categoriesRepository.Object);
+            _categoriesRepository.Setup(x => x.Update(category));
+            _categoriesRepository.Setup(x => x.Find(category.Id)).Returns(null as Category);
+            _uow.Setup(x => x.Save()).Throws(exception);
+
+            //act
+            var actual = _target.PutCategory(category.Id, category);
+
+            //assert
+            (actual as StatusCodeResult)?.StatusCode.Should().Be(404, "Status code is not NoContent");
+            _uow.Verify(x => x.CategoryRepository, Times.Exactly(2));
+            _uow.Verify(x => x.Save(), Times.Once);
+            _categoriesRepository.Verify(x => x.Update(category), Times.Once);
+            _categoriesRepository.Verify(x => x.Find(category.Id), Times.Once);
+        }
+
+        [TestMethod]
+        public void PutCategoryOptimisticExceptionThrowTest()
+        {
+            //arrange
+            var category = _fixture.Create<Category>();
+            var exception = new DbUpdateConcurrencyException();
+
+            _uow.Setup(x => x.CategoryRepository).Returns(_categoriesRepository.Object);
+            _categoriesRepository.Setup(x => x.Update(category));
+            _categoriesRepository.Setup(x => x.Find(category.Id)).Returns(category);
+            _uow.Setup(x => x.Save()).Throws(exception);
+
+            //act & assert
+            _target.Invoking(x => x.PutCategory(category.Id, category))
+                .Should()
+                .Throw<DbUpdateConcurrencyException>();
+
+            _uow.Verify(x => x.CategoryRepository, Times.Exactly(2));
+            _uow.Verify(x => x.Save(), Times.Once);
+            _categoriesRepository.Verify(x => x.Update(category), Times.Once);
+            _categoriesRepository.Verify(x => x.Find(category.Id), Times.Once);
+        }
+
+        private Fixture _fixture = null!;
```

Comment:
you are resettuping this models for each test. So why do you need this fields instead of just variables?